Lib/pdb.py

```diff
@@ -183,15 +183,18 @@ class _ExecutableTarget:
 
 class _ScriptTarget(_ExecutableTarget):
     def __init__(self, target):
-        self._target = os.path.realpath(target)
-
-        if not os.path.exists(self._target):
+        if not os.path.exists(target):
```

Comment:
I'm thinking we should have a one-line change in the constructor, something like:

```suggestion
        self._target = self._safe_realpath(target)
        if not os.path.exists(self._target):
```

Then, `_safe_realpath` would have the logic and a docstring instead of a comment describing why it exists and how it protects against non-existent readlink targets (the code below).

Reply:
And maybe it will help with the test, yeah :)

Reply:
I'm worried `safe_realpath` could be confusing near the current `safe_path(-P)` comment. I'll fixup that comment to reduce confusion.

Reply:
https://github.com/python/cpython/pull/142371#discussion_r2596441617

@jaraco To not change the behavior out of scope of the issue, I'd have to keep the file existence validation invoked prior to calling `_safe_realpath` anyway. `os.path.realpath` does suppress `OSError`, but I'd not mess with that if we can avoid double-checking.

Reply:
Change done.